### PR TITLE
Add `PIsData` and `PlutusType` instances for `PMap` and `PValue`

### PR DESCRIPTION
**Pull request by TotallyNotChase**
Here it is, it works!

We're still missing `PConstant`/`PLift` instance for `PMap` and `PValue` - I can't figure out the error but I have documented it. With that said, I'd argue a `PConstant`/`PLift` instance isn't as important right now.

### CODE REVIEW

**Comment by TotallyNotChase** [comment]
actually I think I can implement the remaining instances manually. Don't merge this yet

**Comment by L-as** [comment]
`PLift a` is necessary for `PBuiltinList a`.

**Comment by TotallyNotChase** [comment]
Never mind, the `PConstant` and `PLift` instances must follow later. Faced some issues. This PR can be merged on its own.

> If further compromise is needed just to get lifting for Value working, then it itself could just be monomorphically special-cased.

So, the instance for `Value` comes naturally and automatically from the deriver if `PMap` has the correct instances. I have tested this. It's just the implementation for `PMap` that I'm stuck on right now. Getting endless missing constraints.....

File: Plutarch/Api/V1.hs [hunk 1]
```diff
@@ -572,10 +571,15 @@ instance PIsDataRepr PDCert where
 
 ---------- AssocMap
 
-newtype PMap (a :: PType) (b :: PType) (s :: S)
-  = PMap (Term s (PBuiltinMap a b))
+newtype PMap (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
```
Comment (L-as) [inline Plutarch/Api/V1.hs:574]: @t1lde why did you make a newtype for this BTW?

**Review by L-as — APPROVED** [review]
LGTM

**Pull request merged**